homeassistant/core.py

```diff
@@ -373,6 +374,35 @@ def from_dict(cls, json_dict):
         return cls(json_dict['entity_id'], json_dict['state'],
                    json_dict.get('attributes'), last_changed, last_updated)
 
+    def as_number(self):
```

Comment:
I have tried to keep the core component agnostic. I would suggest we move this to the helpers.state module